src/session/client-certificate.c

```diff
@@ -74,6 +103,43 @@ read_proc_self_cgroup (size_t *out_length)
   return NULL;
 }
 
+static
+unsigned long long get_proc_pid_start_time (int dirfd)
+{
+  int fd = openat (dirfd, "stat", O_RDONLY | O_CLOEXEC);
+  if (fd < 0)
+    err (EX, "Failed to open /proc/pid/stat");
+
+  char buffer[4096];
+  ssize_t size = read (fd, buffer, sizeof buffer);
```

Comment:
For example: you *don't* check `EINTR` here.  I think that's fine.

I wonder if you want a "read single newline-terminated file from proc via dirfd and filename" helper.  You could mandate a stack-allocated buffer (and store the return value inside of it as well).

Reply:
Also: fwiw: `/proc/self/stat` comes in at 319 bytes on a random `cat` I just tested.  Do we imagine that this will ever exceed 4096 bytes?  I think we can assume that we're reading the entire thing in one go and treat it as an error if not.  At least it would fail in the "good" direction in that case, and we'd get a chance to fix it.

Reply:
No, /proc/pid/stat is documented API, and at least the first 22 fields can't change type (and thus max length) or order. Of course new fields can appear, but we don't care as we stop parsing after start time.

I added a `read_proc_file()` helper to factor out the read() and error checking.